src/client/client.ts

```diff
@@ -1,42 +1,33 @@
-import * as Backoff from 'backo2';
-import {EventEmitter, ListenerFn} from 'eventemitter3';
-
 declare let window: any;
 const _global = typeof global !== 'undefined' ? global : (typeof window !== 'undefined' ? window : {});
 const NativeWebSocket = _global.WebSocket || _global.MozWebSocket;
 
-import {
-  SUBSCRIPTION_FAIL,
-  SUBSCRIPTION_DATA,
-  SUBSCRIPTION_START,
-  SUBSCRIPTION_SUCCESS,
-  SUBSCRIPTION_END,
-  KEEPALIVE,
-  INIT,
-  INIT_FAIL,
-  INIT_SUCCESS,
-} from './messageTypes';
-import { GRAPHQL_SUBSCRIPTIONS } from './protocols';
-
+import * as Backoff from 'backo2';
+import { EventEmitter, ListenerFn } from 'eventemitter3';
 import isString = require('lodash.isstring');
 import isObject = require('lodash.isobject');
+import { ExecutionResult } from 'graphql';
+
 
-export * from './helpers';
+import MessageTypes from '../common/message-types';
+import { GRAPHQL_WS } from '../common/protocol';
+import { WS_TIMEOUT } from './defaults';
 
-export interface SubscriptionOptions {
+export interface RequestOptions {
   query: string;
   variables?: Object;
   operationName?: string;
   context?: any;
 }
 
-export interface Subscription {
-  options: SubscriptionOptions;
+export interface Request {
+  options: RequestOptions;
   handler: (error: Error[], result?: any) => void;
+  isSubscription: boolean;
```

Comment:
isSubscription shouldn't exist, client should eventually treat subscription just as a request, i want to keep the handling of SUBSCRIPTION message only on server side where they will be eventually deprecated